ui/App/components/Mods/Mod.jsx

```diff
@@ -1,33 +1,143 @@
 import React from 'react';
 
 class Mod extends React.Component {
-    togglePress(e) {
-        e.preventDefault();
-        console.log(this.refs.modName);
-        const node = this.refs.modName;
-        const modName = node.name;
-        this.props.toggleMod(modName);
+    constructor(props) {
+        super(props);
+
+        this.toggleUpdateStatus = this.toggleUpdateStatus.bind(this);
+        this.removeVersionAvailableStatus = this.removeVersionAvailableStatus.bind(this);
+
+        this.state = {
+            newVersionAvailable: false,
+            updateInProgress: false
+        }
+    }
+
+    componentDidMount() {
+        this.checkForNewVersion();
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if(prevProps.mod.version != this.props.mod.version) {
+            this.checkForNewVersion();
+        }
+    }
+
+    checkForNewVersion() {
+        let this_class = this;
```

Comment:
You don't need to change this with ES6 Javascript.

In the Component class you can access it's props directly like `this.props.updateCountAdd`

Reply:
I discovered, that using function() { ... } will override the this-variable inside the function.
Using the lambda-expression () => { ... } for functions, dont override the this-varaible.
... JavaScript is funny xD

Reply:
Ah I see that's good to know, ES6 added a bunch of nice features to Javascript.  It's still Javascript though.. XD